solr/solrj/src/test/org/apache/solr/client/solrj/SolrExampleTests.java

```diff
@@ -571,6 +572,67 @@ public void testGetEmptyResults() throws Exception {
     assertEquals(0, out.get(1).size());
   }
 
+  @Test
+  public void testMatchAllPaging() throws Exception {
+    SolrClient client = getSolrClient();
+
+    // Empty the database...
+    client.deleteByQuery("*:*"); // delete everything!
+    if (random().nextBoolean()) {
+      client.commit();
+    }
+    // Add eleven docs
+    List<SolrInputDocument> docs = new ArrayList<>();
+    final int docsTotal = CommonParams.ROWS_DEFAULT + 1;
+    for (int i = 0; i < docsTotal; i++) {
+      SolrInputDocument doc = new SolrInputDocument();
+      doc.addField("id", "id" + i);
+      doc.addField("name", "doc" + i);
+      doc.addField("price", "" + i);
+      docs.add(doc);
+      if (rarely()) {
+        client.add(docs);
+        client.commit();
+        docs.clear();
+      }
+    }
+    client.add(docs);
+    if (random().nextBoolean()) {
+      client.commit();
+    } else {
+      client.optimize();
+    }
+    final List<String> sorts = Arrays.asList("_docid_", "id", "name", "price", null);
+    Collections.shuffle(sorts, random());
+    final List<Integer> starts =
+        Arrays.asList(0, 1, 2, CommonParams.ROWS_DEFAULT, docsTotal, CommonParams.ROWS_DEFAULT + 2);
+    Collections.shuffle(starts, random());
+    for (String sort : sorts.subList(0, 1 + random().nextInt(sorts.size() - 1))) {
+      for (int start : starts.subList(0, 1 + random().nextInt(starts.size() - 1))) {
+        final SolrQuery query = new SolrQuery("*:*");
+        if (sort != null) {
+          query.setSort(sort, random().nextBoolean() ? SolrQuery.ORDER.asc : SolrQuery.ORDER.desc);
+        }
+        if (start > 0 || random().nextBoolean()) {
+          query.setStart(start);
+        }
+        if (usually()) {
+          query.setRows(CommonParams.ROWS_DEFAULT);
+        }
+        SolrDocumentList results = client.query(query).getResults();
+        assertEquals(docsTotal, results.getNumFound());
+        assertEquals(
+            "page from " + start,
+            Math.max(Math.min(CommonParams.ROWS_DEFAULT, docsTotal - start), 0),
+            results.size());
+        for (SolrDocument doc : results) {
+          assertTrue(doc.containsKey("id") && doc.containsKey("name") && doc.containsKey("price"));
+          break;
```

Comment:
Curious why the `break;` is here? And to consider splitting the `&&` clauses to make any failures easier to diagnose.

```suggestion
          assertTrue(doc.containsKey("id"));
          assertTrue(doc.containsKey("name"));
          assertTrue(doc.containsKey("price"));
```

Reply:
Pardon. With `for(..) {.. break;}` I tried to say if not empty checking the first one, do nothing otherwise. I saves a few micros (I guess) in comparison to looping all five docs. Let it be.